### PR TITLE
Update the Blog

### LINKED ISSUE #9: Implement Blog update functionality
Author should be able to edit the blog while viewing the preview, like in creation
update by making put call to the endpoint `/blog/{blogid}`

### PR DESCRIPTION
**Pull request by ugpec79**
## Description

I have created the functionality of updating the blogs

## Issue Resolved
Fixes #9 

## Checklist

Please make sure to review the following before submitting your PR:

- [ x] I have read the [contribution guidelines](https://github.com/ken1000minus7/Bloggy/blob/master/README.md).
- [ x] I have read the code of conduct.
- [ x] I have reviewed my submission in detail.

### CODE REVIEW

File: src/frontend/src/components/Blog.js [hunk 1]
```diff
@@ -70,6 +69,16 @@ export const Blog = ({ blog }) => {
...
+                <button onClick={()=>{
+                   if(localStorage.getItem("username")){
+                                    navigate(`/update/${blog.id}`)
+                                }
+                                else{
+                                    navigate('/login')
+                                }
+                }} class="bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded">
+                Update The Blog
+                </button>
```
Comment (ken1000minus7) [inline src/frontend/src/components/Blog.js:81]: Hide this button when the logged in user is not the author of the blog

File: src/frontend/src/pages/UpdatePage.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                    play
+                    loop
+                    animationData={writerAnimation}
+                    className="h-[80px] md:h-[150px] sm:h-[100px]"
+                    />
+            </div>
+            <div className="mx-[20px] my-[10px] sm:mx-[10px]">
+                <TextField
+                    variant="outlined"
+                    value={blog.title}
```
Comment (ken1000minus7) [inline src/frontend/src/pages/UpdatePage.js]: the value here should be `title` , you can set the title to be the blog title when making the get req above, this way initially the title displayed would be the original title, the same can be done for content

File: src/frontend/src/pages/UpdatePage.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        return ()=>{
+            window.removeEventListener("resize",handleWidth)
+        }
+    })
+    useEffect(()=>{
+        setLoading(true)
+        axios.get(`${process.env.REACT_APP_API_BASE_URL}/blog/${id}`,{
+            method : "GET"
+        })
+            .then(response=> setBlog(response.data))
```
Comment (ken1000minus7) [inline src/frontend/src/pages/UpdatePage.js]: here u can set the values of `title` and `content`

**Review by ken1000minus7 — CHANGES_REQUESTED** [review]
Great work @ugpec79 , need a few changes though
Also share screenshots of the pages changed

**Comment by ugpec79** [comment]
image

**Comment by ugpec79** [comment]
@ken1000minus7 sir I have made changes and the update button in the blog is set inside the if condition already

**Comment by ugpec79** [comment]
I was not able to run the backend due to some db issues so I have posted the pic of update page but the updation in data is done

**Comment by ugpec79** [comment]
image

This is for the image which will be seen by the author

**Review by ken1000minus7 — APPROVED** [review]
Lgtm gw, you can tell about the issues with db in the project channel if u r still facing them, I'll see if what can be done

**Comment by ugpec79** [comment]
Sir I have re installed everything now it is working fine

**Pull request merged**